lib/spack/spack/build_systems/intel.py

```diff
@@ -0,0 +1,191 @@
+        """Components of the package to install."""
+        return self._components
+
+    @components.setter
+    def components(self, requested_components):
+        """Packages like ``intel-parallel-studio`` contain dozens of
+        components, and some users may want to install a subset of these.
+        This setter allows you to customize which components get installed.
+
+        By default, installs 'ALL' components. To install only MKL and DAAL,
+        for example, you would use:
+
+        .. code-block:: python
+
+           components = ['mkl', 'daal']
+        """
+        matches = []
+
+        for valid in _valid_components():
+            for requested in requested_components:
+                if requested in valid:
+                    matches.append(valid)
+
+        self._components = matches
+
+    @property
+    def global_license_file(self):
+        """Returns the path where a global license file should be stored.
+
+        All Intel software shares the same license, so we store it in a
+        common 'intel' directory."""
+        return join_path(self.global_license_dir, 'intel',
+                         os.path.basename(self.license_files[0]))
+
+    def configure(self, spec, prefix):
+        """Writes the ``silent.cfg`` file used to configure the installation.
+
+        See https://software.intel.com/en-us/articles/configuration-file-format
+        """
+        # Patterns used to check silent configuration file
+        #
+        # anythingpat - any string
+        # filepat     - the file location pattern (/path/to/license.lic)
+        # lspat       - the license server address pattern (0123@hostname)
+        # snpat       - the serial number pattern (ABCD-01234567)
+        config = {
+            # Accept EULA, valid values are: {accept, decline}
+            'ACCEPT_EULA': 'accept',
+
+            # Optional error behavior, valid values are: {yes, no}
+            'CONTINUE_WITH_OPTIONAL_ERROR': 'yes',
+
+            # Install location, valid values are: {/opt/intel, filepat}
+            'PSET_INSTALL_DIR': prefix,
+
+            # Continue with overwrite of existing installation directory,
+            # valid values are: {yes, no}
+            'CONTINUE_WITH_INSTALLDIR_OVERWRITE': 'yes',
+
+            # List of components to install,
+            # valid values are: {ALL, DEFAULTS, anythingpat}
+            'COMPONENTS': ';'.join(self.components),
+
+            # Installation mode, valid values are: {install, repair, uninstall}
+            'PSET_MODE': 'install',
+
+            # Directory for non-RPM database, valid values are: {filepat}
+            'NONRPM_DB_DIR': prefix,
+
+            # Perform validation of digital signatures of RPM files,
+            # valid values are: {yes, no}
+            'SIGNING_ENABLED': 'no',
+
+            # Select target architecture of your applications,
+            # valid values are: {IA32, INTEL64, ALL}
+            'ARCH_SELECTED': 'INTEL64',
```

Comment:
Previously, we were using `ALL` here. I changed it to `INTEL64` in the hopes that it would solve all of my headaches. Apparently some of the Intel components don't work unless both 64 and 32-bit libraries are installed for `libc` and `libstdc++`:
```
$ mkl_link_tool -h
mkl_link_tool: error while loading shared libraries: libstdc++.so.6: wrong ELF class: ELFCLASS64
```
Unfortunately, the fix didn't work. We could pare down the list of `COMPONENTS` to exclude x86 things, but I wonder if anyone will want 32-bit support.